### PR TITLE
Add `Socket::passcred`/`set_passcred` for working with `SO_PASSCRED`.

### PR DESCRIPTION
**Pull request by tobz**
This PR adds the ability to get and set the value of `SO_PASSCRED` on sockets.

Theoretically, FreeBSD [does support](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=234722) the equivalent behavior, but does so via differently named constants/values, so I've opted to constrain this implementation to Linux only for now.

### CODE REVIEW

**Review by Thomasdezeeuw — APPROVED** [review]
Any chance you can add a smoke test to this?

**Comment by tobz** [comment]
Woops, _thought_ I added a test but I had committed to the wrong branch.

Should be all set now.

**Comment by tobz** [comment]
@Thomasdezeeuw Small bump.

Not in a rush, but I have the feeling this might have fallen off your radar after the approval. 😄

**Comment by Thomasdezeeuw** [comment]
> Not in a rush, but I have the feeling this might have fallen off your radar after the approval. 😄

Hadn't fallen off my radar, but $dayjob got busy, then my open source work is the first on the chopping block I'm afraid.

Test looks good, for future reference we have a little `test!` macro to help with this: https://github.com/rust-lang/socket2/blob/21ba6609efe09a7e27d62d2c29ff61bab88b970d/tests/socket.rs#L1326. But merge as is.

Thanks @tobz!

**Comment by tobz** [comment]
Doh! Sorry about the pushiness. Totally understand $dayjob cutting into open source time. 😓 

Thanks again for approving/merging! Is there a general cadence you follow for cutting releases? I'm already using a Git ref at the moment, so I'm fine continuing to do so for a little while... but mostly curious. :)

**Comment by Thomasdezeeuw** [comment]
> Doh! Sorry about the pushiness. Totally understand $dayjob cutting into open source time. 😓

:+1: 

> Thanks again for approving/merging! Is there a general cadence you follow for cutting releases? I'm already using a Git ref at the moment, so I'm fine continuing to do so for a little while... but mostly curious. :)

Not really, I mostly make releases on request. We don't see too many changes

**Pull request merged**
